test case improvement

yaojingguo opened the pull request:
1. Remove duplicate function calls in TestEnumerateIsolations
2. Correct the comment for enumerateIsolations

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3521)

<!-- Reviewable:end -->

tamird commented on kv/txn_correctness_test.go:
```diff
@@ -280,7 +280,7 @@ var (
 // each possible combination of such transaction isolations.
 func enumerateIsolations(numTxns int, isolations []roachpb.IsolationType) [][]roachpb.IsolationType {
-	// Use a count from 0 to pow(# isolations, numTxns) and examine
+	// Use a count from 0 to pow(#isolations, numTxns)-1 and examine
```
I'd leave this space after the `#`

tamird commented:
lgtm

yaojingguo commented:
A new commit which leaves the space after the `#`.

tamird commented on kv/txn_correctness_test.go:
```diff
@@ -310,15 +310,17 @@ func TestEnumerateIsolations(t *testing.T) {
-	if !reflect.DeepEqual(enumerateIsolations(3, bothIsolations), expIsolations) {
-		t.Errorf("expected enumeration to match %s; got %s", expIsolations, enumerateIsolations(3, bothIsolations))
+	enum := enumerateIsolations(3, bothIsolations)
+	if !reflect.DeepEqual(enum, expIsolations) {
```
can you make this `if enum := ...; !reflect.DeepEqual(...) {` and the same below? sorry, I missed it below.

yaojingguo replied:
@tamird, another new commit after including your suggested change.

yaojingguo committed:
test case improvement

1. Remove duplicate function calls in TestEnumerateIsolations
2. Correct the comment for enumerateIsolations

tamird commented:
thanks!
